### PR TITLE
[TTAHUB-1641] Auto generate logical diagram

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
To remove the manual process of maintaining the logical data model, it has now been automatically generated. This is done either by running yarn ldm or yarn docker:ldm. It is also automatically by running migrations with yarn db:migrate or yarn docker:db:migrate 

In audiation to automatically generating the model, the process identifies discrepancies between the implemented database and the implemented models. When discrepancies are identified, it highlights them and places a issue description in the file. A test is implemented to aid in reporting these discrepancies do CI will fail when they are present.

The generation of the png/svg has been removed from this process and will be solved via another JIRA/method. Only the encoded and plantuml files are generated.

## How to test
Make a change to either the database or models, then run the ldm tool either directly or by running migration. Note that the ldm files under docs will have been modified.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1641

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — APPROVED** [review]
Great!

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good although I wasn't able to get the docker image to run. I did see changes to the schema files under docs.

File: src/migrations/20230424000000-correct-model-db-misalignment.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      await queryInterface.sequelize.query(
+        `
+        SELECT "ZAFSetTriggerState"(null, null, null, 'ENABLE');
+        `,
+        { transaction },
+      );
+    },
+  ),
+  down: async (queryInterface) => {
+    await queryInterface.sequelize.transaction((transaction) => (queryInterface.removeColumn('Goals', 'rtrOrder', { transaction })));
```
Comment (hardwarehuman) [inline src/migrations/20230424000000-correct-model-db-misalignment.js:110]: This vestigial `down` needs to be removed

File: src/migrations/20230424000000-correct-model-db-misalignment.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        { transaction },
+      );
+      // clean up orphan tables
+      await Promise.all(['ObjectiveTemplateRoles', 'DisconnectedGoals'].map(async (table) => {
+        await queryInterface.sequelize.query(
+          ` SELECT "ZAFRemoveAuditingOnTable"('${table}');`,
+          { raw: true, transaction },
+        );
+        // Drop old audit log table
+        await queryInterface.dropTable(`ZAL${table}`, { transaction });
```
Comment (hardwarehuman) [inline src/migrations/20230424000000-correct-model-db-misalignment.js:93]: Where was `ObjectiveTemplateRoles` removed? I don't see it in `main`

**Review by hardwarehuman — CHANGES_REQUESTED** [review]

File: src/migrations/20230424000000-correct-model-db-misalignment.js (hunk 2 above)
Comment (hardwarehuman) [reply src/migrations/20230424000000-correct-model-db-misalignment.js:93]: Wait, we're not trying to remove the table, I see. You can ignore this one.

**Pull request merged**